appservice/src/SiteWrapper.ts

```diff
@@ -210,7 +210,13 @@ export class SiteWrapper {
         } else if (await FileUtilities.isDirectory(fsPath)) {
             createdZip = true;
             this.log(outputChannel, localize('zipCreate', 'Creating zip package...'));
-            zipFilePath = await FileUtilities.zipDirectory(fsPath);
+            const zipDeployConfig: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration('zipDeploy');
```

Comment:
By hard-coding the setting names in this npm package, you're essentially forcing each extension to add these settings to their package.json. What if instead the `deploy` method takes in an optional "zipOptions" parameter that has the globPattern and ignorePattern

Reply:
If an extension doesn't define the settings, the patterns will just be "undefined"; when passed to the actual zip method, the glob pattern will be defaulted to '**/*' and ignore pattern remains undefined which is also default.